osquery/tables/system/windows/drivers.cpp

```diff
@@ -173,79 +144,135 @@ Status getDeviceProperty(const device_infoset_t& infoset,
   return Status();
 }
 
-static inline std::string getDriverImagePath(const std::string& service_key) {
+std::string getDriverImagePath(const std::string& service_key) {
   SQL sql("SELECT data FROM registry WHERE path = '" + service_key +
           "\\ImagePath'");
-  if (sql.rows().size() == 1) {
-    auto path = sql.rows().at(0).at("data");
-    if (!path.empty()) {
-      CHAR systemRoot[MAX_PATH] = {0};
-      GetSystemDirectory(systemRoot, MAX_PATH);
-      return systemRoot +
-             boost::regex_replace(path, boost::regex("^.*[Ss]ystem32"), "");
-    }
+
+  if (sql.rows().empty() || sql.rows().at(0).count("data") == 0) {
+    return "";
   }
-  return "";
+
+  auto path = sql.rows().at(0).at("data");
+  if (path.empty()) {
+    return "";
+  }
+
+  char systemRoot[MAX_PATH] = {0};
+  GetSystemDirectory(systemRoot, MAX_PATH);
+  return systemRoot +
```

Comment:
I know this is old code, but I see a possibility for bugs here. I assume what this is doing is replacing `%SYSTEMROOT%` or something similar with the actual system root.

* Why don't we search for the actual variable names as opposed to `system32`? This will fail in case the path has another `system32` later on since this regex matching is greedy.
* Are we sure this will always fall on the system32 directory?
* Finally can you put a comment here explaining what this does, mostly for non-Windows people it's hard to  understand and since you're touching this code that would be nice.

Reply:
> Why don't we search for the actual variable names as opposed to system32?
Not sure what you mean here. You have the right idea though, we're getting the system root from Windows, very commonly `C:\Windows\system32`, and pre-pending it to driver image paths. The reason for needing this is it's left to the driver developer to provide the image path, and they're not consistent at all. Below is an example of some of them:
```
...
\SystemRoot\System32\drivers\BasicDisplay.sys
\SystemRoot\System32\drivers\vmbusr.sys
System32\drivers\volume.sys
system32\drivers\tsusbhub.sys
...
```
Continuing to your other point, it's not necessarily the case, I believe, that the driver image path live in `system32`, though this is the common practice and I believe recommended by MS. Even still, we can update the regex to have that be optional.